internal/s3storage/s3storage.go

```diff
@@ -0,0 +1,84 @@
+package s3storage
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/lunarway/release-manager/internal/artifact"
+	"github.com/lunarway/release-manager/internal/log"
+)
+
+type Service struct {
+	bucketName string
+	logger     *log.Logger
+	s3client   *s3.S3
+}
+
+func Initialize(logger *log.Logger) (*Service, error) {
```

Comment:
Could this be `New` instead of `Initialize`? As it returns the allocated struct eg. instead of initializing a global value.

Reply:
Yeah, just wanted to tell the world that it's actually doing stuff - that is creating a bucket in the cloud. You sure you like `New` better considering that?

Reply:
I understand the intention. Another approach would be to separate it out into two calls. A `New` function and `Initialize` method.

But what ever you like :) We can keep it as is.